### ISSUE TITLE
Authentication Angular UI sign-up component throws exception when usernameAlias is set to email and form is not filled out before submitting

### ISSUE THREAD

**Issue by oliverschenk** [aws-amplify/amplify-js#7636]
**Describe the bug**
This relates to an Angular project with AWS Amplify and the Authentication component for sign-in and sign-up. I created a bare bones project with nothing other than `<amplify-authenticator usernameAlias="email"></amplify-authenticator>` in app.component.html and the Amplify configuration described below. 

When the **sign-in** form is submitted without completing any of the fields it shows an error message "Username cannot be empty" as would be expected. However, on the **sign-up** form, if none of the fields are completed then an exception in thrown.

**To Reproduce**

Please find the repo for the below steps if you want the complete thing. You'll have to initialise your own AWS Amplify configuration though.

https://github.com/oliverschenk/aws-amplify-username-alias

```
$ npx create-nx-workspace
npx: installed 190 in 18.988s
? Workspace name (e.g., org name)     my
? What to create in the new workspace angular           [a workspace with a single Angular application]
? Application name                    amplify-usernamealias
? Default stylesheet format           CSS
? Default linter                      ESLint [ Modern linting tool ]
? Use Nx Cloud? (It's free and doesn't require registration.) No
Creating a sandbox with Nx...
added 190 packages from 152 contributors and audited 191 packages in 12.081s

24 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

new my --no-interactive --preset=angular --appName=amplify-usernamealias --style=css --linter=eslint --no-nxCloud --collection=@nrwl/workspace
√ Packages installed successfully.
√ Packages installed successfully.

CREATE jest.config.js
CREATE jest.preset.js
CREATE apps/amplify-usernamealias/tsconfig.editor.json
CREATE apps/amplify-usernamealias/tsconfig.json
CREATE apps/amplify-usernamealias/src/favicon.ico
CREATE apps/amplify-usernamealias/.browserslistrc
CREATE apps/amplify-usernamealias/tsconfig.app.json
CREATE apps/amplify-usernamealias/src/index.html
CREATE apps/amplify-usernamealias/src/main.ts
CREATE apps/amplify-usernamealias/src/polyfills.ts
CREATE apps/amplify-usernamealias/src/styles.css
CREATE apps/amplify-usernamealias/src/assets/.gitkeep
CREATE apps/amplify-usernamealias/src/environments/environment.prod.ts
CREATE apps/amplify-usernamealias/src/environments/environment.ts
CREATE apps/amplify-usernamealias/src/app/app.module.ts
CREATE apps/amplify-usernamealias/src/app/app.component.html
CREATE apps/amplify-usernamealias/src/app/app.component.spec.ts
CREATE apps/amplify-usernamealias/src/app/app.component.ts
CREATE apps/amplify-usernamealias/src/app/app.component.css
CREATE .eslintrc.json
CREATE apps/amplify-usernamealias/.eslintrc.json
CREATE apps/amplify-usernamealias/jest.config.js
CREATE apps/amplify-usernamealias/src/test-setup.ts
CREATE apps/amplify-usernamealias/tsconfig.spec.json
CREATE apps/amplify-usernamealias-e2e/.eslintrc.json
CREATE apps/amplify-usernamealias-e2e/cypress.json
CREATE apps/amplify-usernamealias-e2e/tsconfig.e2e.json
CREATE apps/amplify-usernamealias-e2e/tsconfig.json
CREATE apps/amplify-usernamealias-e2e/src/fixtures/example.json
CREATE apps/amplify-usernamealias-e2e/src/integration/app.spec.ts
CREATE apps/amplify-usernamealias-e2e/src/plugins/index.js
CREATE apps/amplify-usernamealias-e2e/src/support/app.po.ts
CREATE apps/amplify-usernamealias-e2e/src/support/commands.ts
CREATE apps/amplify-usernamealias-e2e/src/support/index.ts
UPDATE angular.json
UPDATE package.json
UPDATE .vscode/extensions.json
UPDATE nx.json
√ Packages installed successfully.
    Successfully initialized git.
CREATE my/nx.json
CREATE my/tsconfig.base.json
CREATE my/package.json
CREATE my/README.md
CREATE my/.editorconfig
CREATE my/.gitignore
CREATE my/.prettierignore
CREATE my/.prettierrc
CREATE my/angular.json
CREATE my/decorate-angular-cli.js
CREATE my/tools/tsconfig.tools.json
CREATE my/tools/generators/.gitkeep
CREATE my/.vscode/extensions.json
CREATE my/apps/.gitkeep
CREATE my/libs/.gitkeep

$ cd my

$ npm install aws-amplify @aws-amplify/ui-angular
npm WARN deprecated buffer@4.9.1: This version of 'buffer' is out-of-date. You must update to v4.9.2 or newer
npm WARN @angular-eslint/eslint-plugin@1.0.0 requires a peer of eslint-plugin-import@* but none is installed. You must install peer dependencies yourself.
npm WARN @angular-eslint/eslint-plugin@1.0.0 requires a peer of eslint-plugin-jsdoc@* but none is installed. You must install peer dependencies yourself.
npm WARN @angular-eslint/eslint-plugin@1.0.0 requires a peer of eslint-plugin-prefer-arrow@* but none is installed. You must install peer dependencies yourself.
npm WARN react-native-get-random-values@1.5.1 requires a peer of react-native@>=0.56 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\webpack-dev-server\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\watchpack-chokidar2\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules\rollup\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.1 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ @aws-amplify/ui-angular@0.4.17
+ aws-amplify@3.3.14
added 317 packages from 410 contributors, removed 1 package and audited 2270 packages in 88.385s

110 packages are looking for funding
  run `npm fund` for details
  
$ amplify configure
Follow these steps to set up access to your AWS account:

Sign in to your AWS administrator account:
https://console.aws.amazon.com/
Press Enter to continue

Specify the AWS Region
? region:  ap-southeast-2
Specify the username of the new IAM user:
? user name:  amplify-ewr6C
Complete the user creation using the AWS console
https://console.aws.amazon.com/iam/home?region=ap-southeast-2#/users$new?step=final&accessKey&userNames=amplify-ewr6C&permissionType=policies&policies=arn:aws:iam::aws:policy%2FAdministratorAccess
Press Enter to continue

Enter the access key of the newly created user:
? accessKeyId:  ********************
? secretAccessKey:  ****************************************
This would update/create the AWS Profile in your local machine
? Profile Name:  amplify-usernamealias-test

Successfully set up the new user.

$ amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project usernamealias
? Enter a name for the environment dev
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using angular
? Source Directory Path:  src
? Distribution Directory Path: dist/apps/amplify-usernamealias
? Build Command:  npm.cmd run-script build
? Start Command: ng serve
Using default provider  awscloudformation


For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

? Do you want to use an AWS profile? Yes
? Please choose the profile you want to use amplify-usernamealias-test
Adding backend environment dev to AWS Amplify Console app: d1wowbtwnbdra5
/ Initializing project in the cloud...
...
√ Successfully created initial AWS cloud resources for deployments.
√ Initialized provider successfully.
Initialized your environment successfully.

Your project has been successfully initialized and connected to the cloud!

$ amplify add auth
Using service: Cognito, provided by: awscloudformation

 The current configured provider is Amazon Cognito.

 Do you want to use the default authentication and security configuration? Manual configuration
 Select the authentication/authorization services that you want to use: User Sign-Up & Sign-In only (Best used with a cloud API only)
 Please provide a friendly name for your resource that will be used to label this category in the project: usernamealias5feb3cfd5feb3cfd
 Please provide a name for your user pool: usernamealias5feb3cfd_userpool_5feb3cfd
 Warning: you will not be able to edit these selections.
 How do you want users to be able to sign in? Email
 Do you want to add User Pool Groups? No
 Do you want to add an admin queries API? No
 Multifactor authentication (MFA) user login options: OFF
 Email based user registration/forgot password: Enabled (Requires per-user email entry at registration)
 Please specify an email verification subject: Your verification code
 Please specify an email verification message: Your verification code is {####}
 Do you want to override the default password policy for this User Pool? No
 Warning: you will not be able to edit these selections.
 What attributes are required for signing up? Email, Name, Phone Number (This attribute is not supported by Facebook, Login With Amazon.)
 Specify the app's refresh token expiration period (in days): 30
 Do you want to specify the user attributes this app can read and write? Yes
 Specify read attributes: Email, Name, Phone Number
 Specify write attributes:
 Do you want to enable any of the following capabilities?
 Do you want to use an OAuth flow? No
? Do you want to configure Lambda Triggers for Cognito? No
Successfully added auth resource usernamealias5feb3cfd5feb3cfd locally

$ amplify push
√ Successfully pulled backend environment dev from the cloud.

Current Environment: dev

| Category | Resource name                 | Operation | Provider plugin   |
| -------- | ----------------------------- | --------- | ----------------- |
| Auth     | usernamealias5feb3cfd5feb3cfd | Create    | awscloudformation |
? Are you sure you want to continue? Yes
- Updating resources in the cloud. This may take a few minutes...
...
√ All resources are updated in the cloud
```
```
polyfills.ts

(window as any).global = window;
```
```
app.module.ts:

import {BrowserModule} from "@angular/platform-browser";
import {NgModule} from "@angular/core";
import {AppComponent} from "./app.component";

import {AmplifyUIAngularModule} from "@aws-amplify/ui-angular";
import Amplify from "aws-amplify";
import awsconfig from "src/aws-exports";

Amplify.configure(awsconfig);

@NgModule({
  declarations: [AppComponent],
  imports: [AmplifyUIAngularModule, BrowserModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
```
```
app.component.html:

<amplify-authenticator usernameAlias="email"></amplify-authenticator>
```
```

$ ng serve
Your global Angular CLI version (11.1.1) is greater than your local version (11.0.7). The local Angular CLI version is used.

To disable this warning use "ng config -g cli.warnings.versionMismatch false".

> nx run amplify-usernamealias:serve
Compiling @aws-amplify/ui-angular : module as esm5
√ Browser application bundle generation complete.

Initial Chunk Files       | Names                  |      Size
vendor.js                 | vendor                 |  14.26 MB
polyfills.js              | polyfills              | 484.74 kB
styles.css, styles.js     | styles                 | 344.30 kB
main.js                   | main                   |  20.07 kB
runtime.js                | runtime                |   9.14 kB

                          | Initial Total          |  15.10 MB

Lazy Chunk Files          | Names                  |      Size
26.js                     | -                      |  98.69 kB
polyfills-core-js.js      | polyfills-core-js      |  78.72 kB
17.js                     | -                      |  67.82 kB
4.js                      | -                      |  42.93 kB
3.js                      | -                      |  39.92 kB
polyfills-dom.js          | polyfills-dom          |  38.52 kB
0.js                      | -                      |  27.04 kB
5.js                      | -                      |  22.95 kB
2.js                      | -                      |  22.80 kB
1.js                      | -                      |  19.83 kB39mm
6.js                      | -                      |  16.90 kB
8.js                      | -                      |  16.88 kB
13.js                     | -                      |  16.69 kB
7.js                      | -                      |  16.21 kB
shadow-css-360bb6da-js.js | shadow-css-360bb6da-js |  15.87 kB
10.js                     | -                      |  13.10 kB
14.js                     | -                      |  11.64 kB
9.js                      | -                      |  11.18 kB
polyfills-css-shim.js     | polyfills-css-shim     |  10.51 kB
common.js                 | common                 |  10.48 kB
11.js                     | -                      |  10.45 kB
15.js                     | -                      |   8.32 kB
12.js                     | -                      |   7.99 kB
20.js                     | -                      |   6.66 kB
19.js                     | -                      |   5.82 kB
27.js                     | -                      |   4.75 kB
21.js                     | -                      |   3.75 kB
25.js                     | -                      |   3.74 kB
28.js                     | -                      |   3.24 kB
29.js                     | -                      |   2.95 kB
22.js                     | -                      |   2.60 kB
16.js                     | -                      |   1.95 kB
24.js                     | -                      |   1.82 kB
23.js                     | -                      |   1.71 kB
18.js                     | -                      |   1.42 kB

Build at: 2021-01-28T06:54:25.220Z - Hash: 36776af0b5f646e66c21 - Time: 19460ms

** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **


√ Compiled successfully.
√ Browser application bundle generation complete.

Initial Chunk Files   | Names  |      Size
vendor.js             | vendor |  14.26 MB
styles.css, styles.js | styles | 344.30 kB

38 unchanged chunks

Build at: 2021-01-28T06:54:32.165Z - Hash: 0785326af20642f440ac - Time: 5879ms

√ Compiled successfully.
```

![7](https://user-images.githubusercontent.com/17871110/106104539-8aa7ef80-617d-11eb-82ae-b05be706a9d6.PNG)
![8](https://user-images.githubusercontent.com/17871110/106104631-ac08db80-617d-11eb-95ce-947274711621.PNG)
![9](https://user-images.githubusercontent.com/17871110/106104643-b034f900-617d-11eb-8c21-298596c8bc87.PNG)

**Expected behavior**
I would expect that if no fields are filled in on the sign-up form, even with `usernameAlias="email"`, that it would shown an error message such as "Username cannot be empty" or "Email cannot be empty".

**Comment by wlee221** [comment]
Thanks for submitting this issue and the repro! This is likely a regression from #7408 -- I'll look into this.

**Comment by wlee221** [comment]
Looks like we should assign empty string here if email field was empty: 
https://github.com/aws-amplify/amplify-js/blob/ae2e1cbfa7407eb810eae1b71ae8ab795d29a542/packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx#L126-L130

I'll make a pr for this soon.

**Comment by wlee221** [comment]
@oliverschenk, fix (#7663) has been published in the latest release. Please let us know if the issue persists after upgrading `@aws-amplify/ui-components` and `aws-amplify`. Thanks!

**Comment by oliverschenk** [comment]
Works now, thanks!
